### DISCUSSION TITLE
Small discrepancy between visits tracked by hosted Plausible and the self hosted version

### DISCUSSION THREAD

**Discussion by vitobotta** [plausible/analytics#2600] [Self-Hosted Support]
I have both self hosted and hosted Plausible configured for a couple of websites (until I decide which one to keep), and I have noticed that the hosted version tracks a few more visits and page views than the self hosted one.

They are both behind a Cloudflare worker (due to ad blockers) with the same configuration, same time zone (since time zone could explain some difference) etc. 

Is there any known explanation to this difference? Thanks

**Comment by metmarkosaric** [comment]
hi @vitobotta! if everything else is the same, then the difference comes from our infrastructure. we put a lot of effort into being highly available, being able to handle large volumes of traffic at any time and loading fast no matter where your visitors are based in the world. you can see the differences between our managed hosting and the self-hosted code here https://plausible.io/self-hosted-web-analytics

**Reply by vitobotta** [reply]
Hi! Thanks for the quick response. The traffic is actually very low (so far it's around 170 visits in 13 hours, so not much, and I have uptime monitoring with 20s frequency to get alerts right away if the self hosted version is down, which hasn't happened.

**Reply by metmarkosaric** [reply]
np! a server might be up but that's not the same as you counting all visitors that land on your site during that time. things such as server specs and a global cdn make a difference

**Comment by ukutaht** [comment]
I wonder how big the difference is?

There are many variables so I can't be sure. My first hunch is that it may have to do with the script loading time. If you have a very short pageview (<1s), then being able to load and fire off a pageview very quickly will make a difference in whether that visit is captured or not.

It could be because the CF worker has to go all the way to your origin server to get the script whereas in our case the script is served from >100 points of presence from around the world, reducing the time it takes to load the script and send a pageview.

**Reply by vitobotta** [reply]
I am based in Finland and the server is in Germany, ping is about 23ms. I don't think that would make much difference?

**Reply by vitobotta** [reply]
@ukutaht I just realized that part of my reply isn't relevant (the fact that I am myself based in Finland). I understand that the Cloudflare worker fetching the script from my origin server might be slower than serving directly from CDN, but the script is actually cached by the CDN. I have tested in various ways and I don't get requests for the script to my origin server.
